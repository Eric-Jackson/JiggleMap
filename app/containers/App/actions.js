/*
 *
 * App actions
 *
 */

import {
  LOAD_PROJECT,
  LOAD_PROJECT_ERROR,
  LOAD_PROJECT_SUCCESS,
  ADD_ENTITY,
  SET_SIDEBAR_ITEM,
} from './constants';

export function loadProject() {
  return {
    type: LOAD_PROJECT,
  };
}

export function loadProjectError(error) {
  return {
    type: LOAD_PROJECT_ERROR,
    error,
  };
}

export function loadProjectSuccess() {
  return {
    type: LOAD_PROJECT_SUCCESS,
  };
}

export function addEntity(path, meta) {
  return {
    type: ADD_ENTITY,
    entity: {
      type: meta.format.type,
      id: meta.id,
      name: meta.name,
      description: meta.description,
      icon: null,
      path,
    },
  };
}

export function setSidebarItem(item) {
  return {
    type: SET_SIDEBAR_ITEM,
    item,
  };
}
