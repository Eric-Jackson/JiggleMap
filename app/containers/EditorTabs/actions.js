/*
 *
 * EditorTabs actions
 *
 */

import {
  LOAD,
  SWITCH_TAB,
  NEW_TAB,
  CLOSE_TAB,
  REORDER_TABS,
  RELAY_ACTION_TO_TAB,
  LOAD_TAB,
} from './constants';

export function newTab() {
  return {
    type: NEW_TAB,
  };
}

export function reorderTabs(order) {
  return {
    type: REORDER_TABS,
    order,
  };
}

export function switchTab(id) {
  return {
    type: SWITCH_TAB,
    id,
  };
}

export function closeTab(id) {
  return {
    type: CLOSE_TAB,
    id,
  };
}

export function relayActionToTab(id, action) {
  return {
    type: RELAY_ACTION_TO_TAB,
    id,
    action,
  };
}

/**
 * Load something in the active tab or create a new tab
 */
export function loadTab(
  kind,
  { title = '', icon = null, meta = {} } = {},
  forceNewTab = false,
  focusTab = false,
) {
  return {
    type: LOAD_TAB,
    kind,
    title,
    icon,
    meta,
    forceNewTab,
    focusTab,
  };
}

/**
 * The loading event
 */
export function load(kind, meta) {
  return {
    type: LOAD,
    kind,
    meta,
  };
}
