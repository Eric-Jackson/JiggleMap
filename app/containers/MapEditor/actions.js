/*
 *
 * MapEditor actions
 *
 */

import {
  LOAD_MAP_ENTITY,
  LOAD_MAIN_MAP,
  LOAD_CONNECTED_MAP,
  EDIT_MAP,
  COMMIT_MAP_EDIT,
  SET_CAMERA_POSITION,
  MOVE_CONNECTION,
  COMMIT_CONNECTION_MOVE,
  RESIZE_VIEWPORT,
  SET_CURRENT_BLOCK,
  MAP_LOADED,
  MOVE_ENTITY,
  COMMIT_ENTITY_MOVE,
} from './constants';

export function loadMapEntity(entity) {
  return {
    type: LOAD_MAP_ENTITY,
    entity,
  };
}

export function loadMainMap({ map, blocksets }) {
  return {
    type: LOAD_MAIN_MAP,
    map,
    blocksets,
  };
}

export function mapLoaded() {
  return {
    type: MAP_LOADED,
  };
}

export function loadConnectedMap({ map, blocksets, direction, offset }) {
  return {
    type: LOAD_CONNECTED_MAP,
    map,
    blocksets,
    direction,
    offset,
  };
}

export function editMap(toolState, start, end, modifiers) {
  return {
    type: EDIT_MAP,
    toolState,
    start,
    end,
    modifiers,
  };
}

export function commitMapEdit() {
  return {
    type: COMMIT_MAP_EDIT,
  };
}

export function setCameraPosition(x, y, z) {
  return {
    type: SET_CAMERA_POSITION,
    x,
    y,
    z,
  };
}

export function moveConnection(connection, x, y) {
  return {
    type: MOVE_CONNECTION,
    connection,
    x,
    y,
  };
}

export function commitConnectionMove(connection) {
  return {
    type: COMMIT_CONNECTION_MOVE,
    connection,
  };
}

export function resizeViewport(width, height) {
  return {
    type: RESIZE_VIEWPORT,
    width,
    height,
  };
}

export function setCurrentBlock(block) {
  return {
    type: SET_CURRENT_BLOCK,
    block,
  };
}

export function moveEntity(id, x, y) {
  return {
    type: MOVE_ENTITY,
    id,
    x,
    y,
  };
}

export function commitEntityMove(id) {
  return {
    type: COMMIT_ENTITY_MOVE,
    id,
  };
}
