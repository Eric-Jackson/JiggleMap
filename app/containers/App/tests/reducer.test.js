import { Map } from 'immutable';

import { loadProject, loadProjectSuccess, loadProjectError } from '../actions';
import projectReducer, { ProjectState } from '../reducer';

describe('projectReducer', () => {
  it('returns the initial state', () => {
    expect(projectReducer(undefined, {})).toMatchSnapshot();
  });

  it('loading the project sets the boolean and resets the state', () => {
    const initialState = ProjectState({
      path: '/path/to/project.json',
      entities: Map({
        key: 'value',
      }),
    });

    const state = projectReducer(initialState, loadProject('/path/to/project.json'));

    expect(state).toEqual(ProjectState({
      error: false,
      loading: true,
      path: '/path/to/project.json',
      map: Map(),
    }));
  });

  it('resets the project state after an error', () => {
    const initialState = ProjectState({
      path: '/path/to/project.json',
      entities: Map({
        key: 'value',
      }),
    });

    const state = projectReducer(initialState, loadProjectError());

    expect(state).toEqual(ProjectState({
      error: true,
      loading: false,
      path: null,
      map: Map(),
    }));
  });

  it('resets the loading state after sucess', () => {
    const initialState = ProjectState({
      path: '/path/to/project.json',
      entities: Map({
        key: 'value',
      }),
      loading: true,
    });

    const state = projectReducer(initialState, loadProjectSuccess());

    expect(state).toEqual(state.set('loading', false));
  });
});
