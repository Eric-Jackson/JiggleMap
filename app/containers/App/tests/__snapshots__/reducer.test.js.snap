// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projectReducer returns the initial state 1`] = `
Object {
  "entities": Object {},
  "error": false,
  "loading": false,
  "sidebarItem": null,
}
`;
